data/core/init.lua

```diff
@@ -744,7 +744,7 @@ end
 function core.open_doc(filename)
   if filename then
     -- try to find existing doc for filename
-    local abs_filename = system.absolute_path(filename)
+    local abs_filename = common.home_encode(system.absolute_path(filename))
```

Comment:
Well, this is contrary to the design decision I made: all the paths should be stored internally using their real path name. 

The home stuff should be only exposed to the user: when user enter a ~ at the beginning it is expanded to the HOME before its usage with the application. Likely when a result is presented to the user any HOME occurrence is substituted with a ~.

With this idea in mind we just need to ensure that doc.abs_filename never contains any ~ but only a real absolute path.

Reply:
Gotcha. Do we want to make it so `system.absolute_path` normalizes `~` to the user's home directory? Or are we good as-is?

Reply:
No, I have already fixed this problem in the commit shown below. The solution was to not use home_encode when storing doc.abs_filename. Please verify it works as intended the fix is in the master branch.

If it is ok we can close this PR as it is no longer needed.